src/action/common/place_nix_configuration.rs

```diff
@@ -28,65 +28,191 @@ impl PlaceNixConfiguration {
         // e.g. https://github.com/NixOS/nix/issues/8444
         #[cfg(not(target_os = "macos"))]
-        settings.insert("auto-allocate-uids".to_string(), "true".to_string());
+        defaults_conf_settings.push(("auto-allocate-uids", "true".into()));
+        let defaults_conf_insert_fragment = defaults_conf_settings
+            .iter()
+            .map(|(s, v)| format!("{s} = {v}"))
+            .collect::<Vec<_>>()
+            .join("\n");
+
+        let nix_conf_insert_position = if Path::new(NIX_CONF).exists() {
+            let existing_nix_conf = tokio::fs::read_to_string(NIX_CONF)
+                .await
+                .map_err(|e| ActionErrorKind::Read(NIX_CONF.into(), e))
+                .map_err(Self::error)?;
+            tracing::trace!("Found existing `/etc/nix/nix.conf`");
+
+            // Find the first line that isn't just `# ...` comments
+            let mut chosen_insert_after = None;
+            // Warn if there seems to be a setting which is a duplicate of one we set.
+            let mut existing_conf_settings = vec![];
+
+            for (index, line) in existing_nix_conf.lines().enumerate() {
+                let line = line.trim();
+                if line.is_empty() || line.starts_with('#') {
+                    continue;
+                } else {
+                    chosen_insert_after = Some(Position::Before {
+                        index,
+                        expected_content: line.to_string(),
+                    });
+                }
+
+                // We only scan one include of depth -- we should make this any depth, make sure to guard for loops
+                if line.starts_with("include") || line.starts_with("!include") {
+                    // Need to read it in if it exists for settings
+                    let path = line
+                        .trim_start_matches("include")
+                        .trim_start_matches("!include")
+                        .trim();
+                    let relative_path = Path::new("/etc/nix/").join(path);
+                    let canonical_path = relative_path.canonicalize().map_err(|e| {
+                        Self::error(ActionErrorKind::Canonicalize(relative_path, e))
+                    })?;
+                    tracing::trace!(path = %canonical_path.display(), "Reading included nix.conf");
+                    let existing_included_conf =
+                        tokio::fs::read_to_string(canonical_path).await.ok(); // If it doesn't exist, ignore it.
+                    if let Some(existing_included_conf) = existing_included_conf {
+                        let lines = existing_included_conf.lines();
+                        for line in lines {
+                            let mut split = line.split('=');
+                            let setting_name = split.next().map(|v| v.trim());
+                            let setting_value = split.next().map(|v| v.trim());
```

Comment:
Do we maybe want to `split_once` instead? Otherwise, it's possible that this could split part of e.g. a cache pubkey (since the pubkey is communicated base64-encoded).